### PR TITLE
Enhance spring boot configuration metadata(port back from 3.3)

### PR DESCRIPTION
**Pull request by oxsean**
## What is the purpose of the change
See #13299

### CODE REVIEW

File: dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/autoconfigure/DubboConfigurationProperties.java [hunk 1]
```diff
@@ -26,110 +27,157 @@
...
     private Map<String, ProviderConfig> providers = new LinkedHashMap<>();
 
+    /**
+     * Multiple configurations for Consumer.
+     */
     private Map<String, ConsumerConfig> consumers = new LinkedHashMap<>();
 
-    private Map<String, ConfigCenterBean> configCenters = new LinkedHashMap<>();
+    /**
+     * Multiple configurations for ConfigCenterBean.
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/autoconfigure/DubboConfigurationProperties.java:162]: `ConfigCenterBean`  change to `ConfigCenter`

File: dubbo-common/src/main/java/org/apache/dubbo/config/TripleConfig.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ * limitations under the License.
+ */
+package org.apache.dubbo.config;
+
+import java.io.Serializable;
+
+/**
+ * Configuration for triple protocol.
+ */
+public class TripleConfig implements Serializable {
```
Comment (CrazyHZM) [inline dubbo-common/src/main/java/org/apache/dubbo/config/TripleConfig.java:24]: We should not add new config content in the 3.2 branches.

File: dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/autoconfigure/DubboConfigurationProperties.java [hunk 3]
```diff
@@ -298,64 +362,23 @@ public void setTracings(Map<String, TracingConfig> tracings) {
...
-        public void setMode(ConfigMode mode) {
-            this.mode = mode;
-        }
-    }
-
-    static class Scan {
+    /**
+     * Configuration for rpc.
+     */
+    public static class RpcConfig {
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/autoconfigure/DubboConfigurationProperties.java:368]: We should not add new config content in the 3.2 branches.

File: dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
+  "properties": [
+    {
+      "name": "dubbo.enabled",
+      "description": "Whether enable autoconfiguration of dubbo, default value is <code>true</code>.",
+      "type": "java.util.Set<java.lang.String>"
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:6]: The type should be `java.lang.Boolean`.

File: dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+- For standard configuration options, add javadoc to the corresponding configuration classes in `dubbo-common`.
+- For non-standard configuration options, there are unnecessary to add nested classes. add them directly to `additional-spring-configuration-metadata.json`.
+
+## Configuration Javadoc Guideline
+
+1. For noun-type configuration options, use "The xxx" format for comments.
+2. For boolean-type configuration options, use "Whether to xxx" format and add ", default value is &lt;code&gt;true&lt;/code&gt;" at the end.
+3. For configuration options with longer comments, use multi-line comments, with a clear summary in the first line.
+4. All comments should end with a period.
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/README.md:25]: I think these should be moved to Javadoc of  DubboConfigurationProperties、DubboMetadataGenerateAutoConfiguration.

File: dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+Configuration beans of dubbo are located in the `dubbo-common` module, and it's not suitable to add spring boot dependencies. 
+However, spring configuration metadata generation relies on read javadoc from the source code and cannot use the @NestedConfigurationProperty annotation. 
+This leads to missing comments and a lack of nested configuration options. Therefore, we use an independent module to copy the code and generate metadata.
+
+## Principles
+
+1. Copy classes under `org/apache/dubbo/config` from `dubbo-common` to the `generated-sources` directory.
+2. Replace `@Nest` with `@NestedConfigurationProperty`.
+3. Use an annotation-only option to compile and generate `spring-configuration-metadata.json`.
+
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-compatible/autoconfigure/README.md:14]: These should be moved to pr description. And add some reason for the plugin.

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

**Comment by AlbumenJ** [comment]
We can enhance this feature in 3.3. In my opinion, we are no need to fix it in 3.2.x

**Pull request closed without merging**